package.json

```diff
@@ -84,14 +84,14 @@
     "lint:md": "yarn --silent lint:md:ci --fix",
     "lint:md:ci": "prettylint '**/*.{md,yml,yaml}' --ignore-path .gitignore",
     "postinstall": "opencollective postinstall && yarn build",
-    "publish": "yarn build-clean && yarn build && lerna publish --silent",
+    "publish": "yarn build-clean && yarn typecheck && yarn build && lerna publish --silent",
```

Comment:
@rubennorte is this fine? We need to ensure the typescript build is ran so the type definition files are generated.

If you _don't_ use this script, anyway else we can ensure it runs before publishing?

Reply:
Possibly cleaner alternative would be doing a tsc `noEmit` for `typecheck` and a tsc `emitDeclarationOnly`  for `build`

Reply:
you cannot use `noEmit` with references. And even if it did, that doesn't make sense to me. Why do the same work twice? You cannot emit without typechecking anyways, so it's just more work for what I'd say is zero gain.

Keeping the typecheck out of `build` means we can run tests on CI even if types fail. Different signals

Reply:
Okay, I get why that wouldn't make sense for us. Works better in projects where a compile step is only needed for production, not for development/testing.

Reply:
@SimenB if that command generates definitions it should be part of the `build` script. We don't always use the `publish` command but we always run the build.

Reply:
Alright, that seems fair :) Will slow down CI a bit since we run `build` as `postinstall`, but it's not too bad